LJMR_MAIN to Main Merge

lexdyel-mendez opened the pull request:
Implemented REST API with Flask and PyMongo. 

REST API Functiong Implemented for the GDP Page are INSERT, DELETE , PUT and GET. They all follow a main -> handler -> dao schema which can be implemented for the rest of the collections with ease.

lexdyel-mendez committed:
Conda Environment

Created the conda env. with Flask and pymongo. The env can be found as a yml, and it can be imported using the instructions in conda documentation.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Dummy Flask

Flask is now running, however its still not connected to mongodb.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Connected to MongoAtlas

Was able to connect the Flask with the MongoDB atlas and created a new collection called user under development.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Started Reformatting

Organizing files since now it is understood weel how to connect to the DB.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
REST API Functions for GDP

Programmed the functions for adding, and obtaining results for the GDP from the db are.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Delete GDP API

Added the Delete one function to the GDP page in the REST API. The same works by filtering through the year, and returns a deletion status if succeeded.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Update GDP API

Added the Update function to the GDP database. All 4 basic functions have been implemented, now it would be to expand them

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Update GDP API

Added the Update function to the GDP database. All 4 basic functions have been implemented, now it would be to expand them

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Update GDP API

Added the Update function to the GDP database. All 4 basic functions have been implemented, now it would be to expand them

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Merge remote-tracking branch 'origin/ljmr_main' into ljmr_main

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

# Conflicts:
#	backend/handlers/gdp.py

lexdyel-mendez committed:
Dummy Flask

Flask is now running, however its still not connected to mongodb.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Added the Postman Collection

The postman collection is the test routes that are being used to test the REST api of the page.

lexdyel-mendez committed:
Merge remote-tracking branch 'origin/ljmr_main' into ljmr_main

lexdyel-mendez committed:
Added Heroku Required Docs

Since Heroku requires the Procfile generated and the requirements.txt, these were generated and added. Procfile was generated using gunicorn and the requirements.txt with pip freeze.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Fixed Requirements for Heroku

The requirments.txt has some issues, it was giving paths instead of versions. Used the command `conda list -e > requirements.txt` to generate another txt.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Fixed Requirements for Heroku

The requirments.txt has some issues, it was giving paths instead of versions. Used the command `conda list -e > requirements.txt` to generate another txt.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Fixed Requirements for Heroku

The requirments.txt has some issues, it was giving paths instead of versions. Used the command `conda list -e > requirements.txt` to generate another txt.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Merge remote-tracking branch 'origin/ljmr_main' into ljmr_main

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

# Conflicts:
#	requirements.txt

lexdyel-mendez committed:
Merge remote-tracking branch 'origin/ljmr_main' into ljmr_main

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

# Conflicts:
#	requirements.txt

lexdyel-mendez committed:
Merge remote-tracking branch 'origin/ljmr_main' into ljmr_main

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

# Conflicts:
#	requirements.txt

lexdyel-mendez committed:
Merge remote-tracking branch 'origin/ljmr_main' into ljmr_main

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

# Conflicts:
#	requirements.txt

lexdyel-mendez committed:
Merge remote-tracking branch 'origin/ljmr_main' into ljmr_main

AdeanColon committed:
Create react app

juandiaz28 commented on backend/config/mongo_config.py:
Check password and user visibility

lexdyel-mendez committed:
Merge branch 'React-setup' into ljmr_main

# Conflicts:
#	.idea/Centro-Capital.iml
#	.idea/misc.xml

lexdyel-mendez committed:
Merged with Front-End

Now the branch has files of both backend and front-end.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Minor Fix to Procfile

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Connected Front-end w Backend

The Welcome.js page is currently the only one pulling from the flask backend, to show a front-end page. ITs very basic, but it can be implemented to the rest of the pages.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Re added Procfile

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Fixed File Locations

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Fixed File Locations

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Fixed Profcile

Now it points to main inside of the backend folder.

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Tried Fix for Heroku deployment error H10

Tried a npm re installation and fixed some valued on the package.json file

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>

lexdyel-mendez committed:
Re fixed Procfile

It should now read main from subdirectory

Signed-off-by: Lexdyel Mendez Rios <lexdyel.mendez@upr.edu>
